### PR TITLE
Added testing

### PR DESCRIPTION
**Pull request by UnitingDust**
Added Test Class for Profile. Modified RegisterServletTest, TestDataServlet, , PersistentDataStoreTest, and PersistentStorageAgentTest to include testing for profiles.

### CODE REVIEW

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 1]
```diff
@@ -10,7 +10,10 @@
 import org.junit.Assert;
 import org.junit.Test;
 import org.mockito.Mockito;
+
```
Comment (ryanluo) [inline src/test/java/codeu/controller/RegisterServletTest.java]: intentional?

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 2]
```diff
@@ -55,26 +58,36 @@ public void testDoPost_BadUsername() throws IOException, ServletException{
...
+    //Return false when running isUserRegistered() method
+    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(false);
+    
     HttpSession mockSession = Mockito.mock(HttpSession.class);
     Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
 
     registerServlet.doPost(mockRequest, mockResponse);
    
-   //stores the user objects created
+    //Store the User and Profile object that were created
```
Comment (ryanluo) [inline src/test/java/codeu/controller/RegisterServletTest.java]: nit: // Store

File: src/test/java/codeu/controller/RegisterServletTest.java [hunk 3]
```diff
@@ -55,26 +58,36 @@ public void testDoPost_BadUsername() throws IOException, ServletException{
...
-    //returns false if is a new user
-    Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(false);
-    //if the user is not already registered, set up the user accoount
+    ProfileStore mockProfileStore = Mockito.mock(ProfileStore.class);
+    
     registerServlet.setUserStore(mockUserStore);
-
+    registerServlet.setProfileStore(mockProfileStore);
+    
+    //Return false when running isUserRegistered() method
```
Comment (ryanluo) [inline src/test/java/codeu/controller/RegisterServletTest.java]: nit: same

Comment (ryanluo) [reply src/test/java/codeu/controller/RegisterServletTest.java]: as below

File: src/test/java/codeu/model/data/ProfileTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ProfileTest {
+	
+	@Test
+	public void testCreate() {
+		String description = "test";
```
Comment (ryanluo) [inline src/test/java/codeu/model/data/ProfileTest.java]: nit: indentation

File: src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java [hunk 5]
```diff
@@ -147,4 +148,30 @@ public void testSaveAndLoadMessages() throws PersistentDataStoreException {
...
+  public void testSaveAndLoadProfiles() throws PersistentDataStoreException {
+	  UUID idOne = UUID.randomUUID();
+	  String description1 = "test_1";
+	  Profile profile1 = new Profile(idOne, description1);
+	  
+	  UUID idTwo = UUID.randomUUID();
+	  String description2 = "test_2";
+	  Profile profile2 = new Profile(idTwo, description2);
+	  
+	  //Save the Profiles
```
Comment (ryanluo) [inline src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java]: // Save

**Comment by UnitingDust** [comment]
Changes made!

**Pull request merged**